src/main/java/uk/gov/companieshouse/api/accounts/interceptor/OpenTransactionInterceptor.java

```diff
@@ -1,53 +1,54 @@
 import uk.gov.companieshouse.api.accounts.transaction.Transaction;
-import uk.gov.companieshouse.api.accounts.transaction.TransactionManager;
 import uk.gov.companieshouse.api.accounts.transaction.TransactionStatus;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
 
 @Component
 public class OpenTransactionInterceptor extends HandlerInterceptorAdapter {
 
-    @Autowired
-    private TransactionManager transactionManager;
+    private static final Logger LOGGER = LoggerFactory
+        .getLogger(CompanyAccountsApplication.APPLICATION_NAME_SPACE);
 
     /**
-     * Pre handle method to validate the request before it reaches the controller. Check if the url
-     * has an existing transaction and to further check if transaction is open. If transaction is
-     * not found then return 404
+     * Pre handle method to validate the request before it reaches the controller by checking if
+     * transaction's status is open.
      */
     @Override
     @SuppressWarnings("unchecked")
     public boolean preHandle(HttpServletRequest request, HttpServletResponse response,
-            Object handler) {
-        try {
-            Map<String, String> pathVariables = (Map) request
-                    .getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE);
-            String transactionId = pathVariables.get("transactionId");
-            ResponseEntity<Transaction> transaction = transactionManager
-                    .getTransaction(transactionId, request.getHeader("X-Request-Id"));
-            request.setAttribute(AttributeName.TRANSACTION.getValue(), transaction.getBody());
-            return isTransactionIsOpen(transaction);
-        } catch (HttpClientErrorException httpClientErrorException) {
-            response.setStatus(httpClientErrorException.getStatusCode().value());
+        Object handler) {
+        Transaction transaction = (Transaction) request
+            .getAttribute(AttributeName.TRANSACTION.getValue());
+
+        if (transaction == null) {
+            logRequestError(request);
+            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
             return false;
         }
+
+        return TransactionStatus.OPEN.getStatus().equals(transaction.getStatus());
     }
 
     /**
-     * Returns whether transaction is open or not.
+     * Log error.
+     *
+     * @param request
      */
-    private boolean isTransactionIsOpen(ResponseEntity<Transaction> transaction) {
-        return (transaction.getBody() != null && transaction.getBody().getStatus()
-                .equals(TransactionStatus.OPEN.getStatus()));
+    private void logRequestError(HttpServletRequest request) {
+        final Map<String, Object> debugMap = new HashMap<>();
+        debugMap.put("request_method", request.getMethod());
+        debugMap
+            .put("message", "OpenTransactionInterceptor error: no transaction in request session");
```

Comment:
same thing here about a specific error message in a generic method - can again be done inline rather than extracting to a method